### PR TITLE
install: Tarantool build in docker

### LINKED ISSUE #131: Come up with a solution for tt environment distribution
`tt pack` command is supposed to pack full tt environment including tt and tarantool binaries. But these binaries (even static versions) are dependent on libc. This dependency gives an error when invoking on Linux distribution with older libc version:
tt, built on Ubuntu 22.04, run on Centos 7
```
/usr/share/tt/tt 
/usr/share/tt/tt: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by /usr/share/tt/tt)
/usr/share/tt/tt: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by /usr/share/tt/tt)
```
Tarantool static, built on Ubuntu 22.04, run on Centos 7:
```
/tarantool/static-build/tarantool-prefix/bin/tarantool
/tarantool/static-build/tarantool-prefix/bin/tarantool: /lib64/libm.so.6: version GLIBC_2.27' not found (required by /tarantool/static-build/tarantool-prefix/bin/tarantool)
/tarantool/static-build/tarantool-prefix/bin/tarantool: /lib64/libm.so.6: version GLIBC_2.29' not found (required by /tarantool/static-build/tarantool-prefix/bin/tarantool)
/tarantool/static-build/tarantool-prefix/bin/tarantool: /lib64/libc.so.6: version GLIBC_2.28' not found (required by /tarantool/static-build/tarantool-prefix/bin/tarantool)
/tarantool/static-build/tarantool-prefix/bin/tarantool: /lib64/libc.so.6: version GLIBC_2.27' not found (required by /tarantool/static-build/tarantool-prefix/bin/tarantool)
/tarantool/static-build/tarantool-prefix/bin/tarantool: /lib64/libc.so.6: version GLIBC_2.18' not found (required by /tarantool/static-build/tarantool-prefix/bin/tarantool)
/tarantool/static-build/tarantool-prefix/bin/tarantool: /lib64/libc.so.6: version GLIBC_2.32' not found (required by /tarantool/static-build/tarantool-prefix/bin/tarantool)
/tarantool/static-build/tarantool-prefix/bin/tarantool: /lib64/libc.so.6: version GLIBC_2.33' not found (required by /tarantool/static-build/tarantool-prefix/bin/tarantool)
/tarantool/static-build/tarantool-prefix/bin/tarantool: /lib64/libc.so.6: version GLIBC_2.25' not found (required by /tarantool/static-build/tarantool-prefix/bin/tarantool)
/tarantool/static-build/tarantool-prefix/bin/tarantool: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by /tarantool/static-build/tarantool-prefix/bin/tarantool)
```
This issue blocks us from packing and distributing full env between different Linux distributions. Need to explore possible solutions:
- Static build with libc. Gives warnings regarding linking getaddrinfo and other functions.
- Static build with musl libc.
- Use pre-compiled binaries with older version of Glibc, install them from repo in docker image and other Linux distros.

### PR DESCRIPTION
**Pull request by psergee**
Closes #131

### CODE REVIEW

File: cli/install/install.go [hunk 1]
```diff
@@ -337,20 +345,19 @@ func downloadRepo(repoLink string, tag string, dst string,
 	logFile *os.File, verbose bool) error {
 	var err error
 	if tag == "master" {
-		err = util.ExecuteCommand("git", verbose, logFile, dst, "clone",
-			"-j", "18", repoLink,
```
Comment (0x501D) [inline cli/install/install.go:341]: Why `-j n` flag was removed?

File: cli/install/install.go [hunk 2]
```diff
@@ -644,10 +651,98 @@ func copyBuildedTarantool(binPath, incPath, binDir, includeDir, version string,
...
+
+	goTextEngine := engines.NewDefaultEngine()
+	dockerfileText, err := goTextEngine.RenderText(string(tarantoolBuildDockerfile),
+		map[string]string{"uid": currentUser.Uid})
+	if err != nil {
+		return err
+	}
+
+	// Write docker file (rw-rw-r-- permissions).
+	if err = ioutil.WriteFile(filepath.Join(tmpDir, "Dockerfile"), []byte(dockerfileText),
```
Comment (0x501D) [inline cli/install/install.go:678]: Why not use a constant name for the docker file? it is already in use in two places.

File: cli/install/install.go (hunk 1 above)
Comment (psergee) [reply cli/install/install.go:341]: This option is not supported in older git versions, which makes `tt install` non-working on some Linux distributions.

File: cli/docker/docker.go [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+		return err
+	}
+	log.Info("Docker image is built.")
+
+	containerId, err := createContainer(dockerClient, runOptions)
+	if err != nil {
+		return nil
+	}
+	defer func() {
+		log.Debugf("Removing container %s", containerId[:12])
```
[the remark this answers is not in the document]
Comment (psergee) [reply cli/docker/docker.go:157]: Since containerId is a very long hash, I use only first 12 chars from it as docker does.
Log message below - fixed.

**Review by 0x501D — APPROVED** [review]

**Review by better0fdead — CHANGES_REQUESTED** [review]
Hello, thx for patch. Can you please add descriptions to some commits. And also "Part of #"/ "Follow up #".

**Review by better0fdead — APPROVED** [review]

**Pull request merged**